(feat):  Link widget improvment

dobri1408 opened the pull request:
https://github.com/plone/volto/issues/4303

dobri1408 committed:
link widget

dobri1408 committed:
link widget

dobri1408 committed:
link widget

dobri1408 committed:
link widget

dobri1408 committed:
link widget

dobri1408 committed:
fix

dobri1408 committed:
fix

dobri1408 committed:
fix

dobri1408 committed:
fix

dobri1408 committed:
add

dobri1408 committed:
d

dobri1408 committed:
fixes

dobri1408 committed:
d

dobri1408 committed:
d

dobri1408 committed:
d

dobri1408 committed:
improve

dobri1408 committed:
improve

dobri1408 committed:
improve

dobri1408 committed:
improve

dobri1408 committed:
fix

dobri1408 committed:
fix

dobri1408 commented:
How is looking?

https://github.com/user-attachments/assets/cbeab282-2a11-4e28-9ad7-bb9c02dfcaa8

<img width="311" alt="Screenshot 2025-05-28 at 13 09 45" src="https://github.com/user-attachments/assets/794d70ec-cf3d-4b40-bb77-4e072fccebf0" />

dobri1408 committed:
fix

dobri1408 committed:
remove cy

dobri1408 committed:
fix

dobri1408 committed:
Update README.md

dobri1408 committed:
Update README.md

dobri1408 committed:
Update README.md

pnicolli commented on packages/volto/src/components/manage/AnchorPlugin/components/LinkButton/AddLinkForm.jsx:
```diff
@@ -1,324 +1,446 @@
+      .map((item) => ({
+        label: item.title,
+        value: item['@id'],
+        src: `${item['@id']}/@@images/image/preview`,
```
The preview scale is 400x400, we should probably use a smaller one. Probably thumb, or icon.

dobri1408 committed:
Update packages/volto/news/7120.feature

Co-authored-by: Piero Nicolli <pnicolli@users.noreply.github.com>

stevepiercy commented on packages/volto-slate/news/7120.feature:
```diff
@@ -0,0 +1 @@
+(feat): Link widget improvment #7120 @dobri1408
```
@pnicolli correct.

However, this change log is useless other than satisfying the requirement of having an entry. What exactly improved and how? Let's help the reader instead of having them dig into the pull request to determine this. See https://6.docs.plone.org/contributing/index.html#write-a-good-change-log-entry.

dobri1408 committed:
Merge branch 'main' into link-widget

dobri1408 committed:
D

dobri1408 committed:
implement feedback

dobri1408 committed:
rebase

dobri1408 committed:
Improve link widget with enhanced picker features

Enhanced the link picker functionality to include searchable suggestions, content previews, and faster link selection during editing.

dobri1408 committed:
Remove changes

dobri1408 committed:
Update pnpm-lock.yaml

dobri1408 committed:
changelog

dobri1408 committed:
changelog

dobri1408 committed:
fix test

dobri1408 committed:
fix tests

dobri1408 committed:
Prevent image widget link clicks from bubbling

dobri1408 committed:
Fix input selector for image URL field in tests

dobri1408 committed:
Merge branch 'main' into link-widget

dobri1408 committed:
fix css

dobri1408 committed:
fix css

dobri1408 committed:
fix css

dobri1408 committed:
Refactor styles for link form select components

Refactor styles for link form select components, updating class names and layout properties.

dobri1408 committed:
Refactor LinkButton to simplify menu handling

Removed unused calculations for menu dimensions and styles.

dobri1408 committed:
Merge branch 'main' into link-widget

dobri1408 committed:
Update selectors for link form input in tests

dobri1408 committed:
Remove redundant flex-shrink property

dobri1408 committed:
Update selector for book cover image placeholder

sneridagh reviewed (changes requested):
@dobri1408 when you try to edit a link, clicking on the unlink button, it errors.
Also the positioning is weird:

<img width="603" height="238" alt="Image" src="https://github.com/user-attachments/assets/4abb9054-3094-4f83-9c78-ca9359b6384b" />

Could you please write docs for it and cover with a test the use case that is erroring?
For the docs, for example, I don't know what are the default items showing nor which fields are we searching for.

But overall it's great!!
